diff --git a/dist/lib/cmds/run.js b/dist/lib/cmds/run.js
index cfd733b0d62965b6874d8b2824746b0ee8be041e..6be0b9ae75e91cdd0787a868a0788f8e2fc5dd90 100644
--- a/dist/lib/cmds/run.js
+++ b/dist/lib/cmds/run.js
@@ -9,7 +9,7 @@ var __rewriteRelativeImportExtension = (this && this.__rewriteRelativeImportExte
 import { createRequire } from 'node:module';
 import { promisify as p } from 'node:util';
 import { Args, Command, Flags } from '@oclif/core';
-import _csv from 'csv-parse';
+import { parse as _csv } from 'csv-parse';
 import createDebug from 'debug';
 import { CommonRunFlags } from "../cli/common-flags.js";
 const debug = createDebug('commands:run');
diff --git a/dist/lib/util/prepare-test-execution-plan.js b/dist/lib/util/prepare-test-execution-plan.js
index 52236a66d6ecef4b25b1337252435e1f6be03658..d2e2ffa4931f701eb426dfb1c5a24a39e05129fb 100644
--- a/dist/lib/util/prepare-test-execution-plan.js
+++ b/dist/lib/util/prepare-test-execution-plan.js
@@ -2,7 +2,7 @@ import fs from 'node:fs';
 import { createRequire } from 'node:module';
 import path from 'node:path';
 import { promisify as p } from 'node:util';
-import csv from 'csv-parse';
+import { parse as csv } from 'csv-parse';
 import createDebug from 'debug';
 const debug = createDebug('artillery');
 // Used for: loading user-provided .js scripts and esbuild-generated
@@ -109,7 +109,7 @@ async function readPayload(script) {
     return script;
 }
 function transpileTypeScript(entryPoint, outputPath, userExternalPackages) {
-    const esbuild = require('esbuild-wasm');
+    const esbuild = require('esbuild');
     esbuild.buildSync({
         entryPoints: [entryPoint],
         outfile: outputPath,
@@ -140,7 +140,7 @@ function replaceProcessorIfTypescript(script, scriptPath) {
     const newProcessorPath = path.join(processorDir, `dist/${processorFileName}.js`);
     //TODO: move require to top of file when Lambda bundle size issue is solved
     //must be conditionally required for now as this package is removed in Lambda for now to avoid bigger package sizes
-    const esbuild = require('esbuild-wasm');
+    const esbuild = require('esbuild');
     try {
         esbuild.buildSync({
             entryPoints: [actualProcessorPath],
diff --git a/lib/cmds/run.ts b/lib/cmds/run.ts
index 1d7eb6fa98c66aaad4fcba8dde2b4bc5d7162e8e..bf1a23bca61c6a2eb9510f700e3f080391632a4c 100644
--- a/lib/cmds/run.ts
+++ b/lib/cmds/run.ts
@@ -1,7 +1,7 @@
 import { createRequire } from 'node:module';
 import { promisify as p } from 'node:util';
 import { Args, Command, Flags } from '@oclif/core';
-import _csv from 'csv-parse';
+import { parse as _csv } from 'csv-parse';
 import createDebug from 'debug';
 import { CommonRunFlags } from '../cli/common-flags.ts';
 
diff --git a/lib/util/prepare-test-execution-plan.ts b/lib/util/prepare-test-execution-plan.ts
index f8ea361d5a55c8f6ef46a2f19c6f283de35adc09..450d05285575ec441e8eea93d8b15c02a23a9c67 100644
--- a/lib/util/prepare-test-execution-plan.ts
+++ b/lib/util/prepare-test-execution-plan.ts
@@ -2,7 +2,7 @@ import fs from 'node:fs';
 import { createRequire } from 'node:module';
 import path from 'node:path';
 import { promisify as p } from 'node:util';
-import csv from 'csv-parse';
+import { parse as csv } from 'csv-parse';
 import createDebug from 'debug';
 
 const debug = createDebug('artillery');
@@ -175,7 +175,7 @@ function transpileTypeScript(
   outputPath: string,
   userExternalPackages: string[]
 ) {
-  const esbuild = require('esbuild-wasm');
+  const esbuild = require('esbuild');
 
   esbuild.buildSync({
     entryPoints: [entryPoint],
@@ -223,7 +223,7 @@ function replaceProcessorIfTypescript(
 
   //TODO: move require to top of file when Lambda bundle size issue is solved
   //must be conditionally required for now as this package is removed in Lambda for now to avoid bigger package sizes
-  const esbuild = require('esbuild-wasm');
+  const esbuild = require('esbuild');
 
   try {
     esbuild.buildSync({
